test/e2e/runner/load.go

```diff
@@ -95,32 +96,56 @@ func loadGenerate(ctx context.Context, chTx chan<- types.Tx, multiplier int, siz
 func loadProcess(ctx context.Context, testnet *e2e.Testnet, chTx <-chan types.Tx, chSuccess chan<- types.Tx) {
 	// Each worker gets its own client to each node, which allows for some
 	// concurrency while still bounding it.
-	clients := map[string]*rpchttp.HTTP{}
+
+	clients := make([]*rpchttp.HTTP, 0, len(testnet.Nodes))
+
+	for idx := range testnet.Nodes {
+		if testnet.Nodes[idx].Mode == e2e.ModeSeed {
+			continue
+		}
+		client, err := testnet.Nodes[idx].Client()
+		if err != nil {
+			continue
+		}
+
+		clients = append(clients, client)
+	}
+
+	if len(clients) == 0 {
+		panic("no clients to process load")
+	}
+
+	clientRing := ring.New(len(clients))
+	for idx := range clients {
+		clientRing.Value = clients[idx]
+		clientRing = clientRing.Next()
+	}
 
 	var err error
-	for tx := range chTx {
-		node := testnet.RandomNode()
 
-		client, ok := clients[node.Name]
-		if !ok {
-			client, err = node.Client()
-			if err != nil {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case tx := <-chTx:
```

Comment:
```suggestion
		case tx, ok := <-chTx; ok:
```

unless we are OK with getting a zero transaction if/when the transaction channel closes. Maybe that can't happen, in which case ignore this.

Reply:
the channel is created in the function above and is never closed.